zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterDownloader.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.interpreter.remote;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Files;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+
+import org.apache.commons.io.FileUtils;
+import org.apache.zeppelin.interpreter.Constants;
+import org.apache.zeppelin.interpreter.thrift.LibraryMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RemoteInterpreterDownloader {
+  private static final Logger LOGGER = LoggerFactory.getLogger(RemoteInterpreterDownloader.class);
+
+  private static final int MAX_LIBRARY_DOWNLOAD_ATTEMPTS = 3;
+
+  private final RemoteInterpreterEventClient client;
+  private final String interpreter;
+  private final File localRepoDir;
+
+  public RemoteInterpreterDownloader(
+      String interpreter,
+      RemoteInterpreterEventClient client,
+      File localRepoDir) {
+    this.client = client;
+    this.interpreter = interpreter;
+    this.localRepoDir = localRepoDir;
+  }
+
+  public static void main(String[] args) {
+    String zeppelinServerHost = null;
+    int port = Constants.ZEPPELIN_INTERPRETER_DEFAUlT_PORT;
+    String interpreter = null;
+    String localRepoPath = null;
+    if (args.length > 0) {
+      zeppelinServerHost = args[0];
+      port = Integer.parseInt(args[1]);
+      interpreter = args[2];
+      localRepoPath = args[3];
+    }
```

Comment:
It seems that the argument check is not strict because the location where this file is used is fixed. All four arguments seem be mandatory, Wouldn't it be better to add an `assert` to work only with "`args.length` == 4"?

Reply:
You are right. I have changed the implementation to properly evaluate the mandatory arguments.